rg_instructor_analytics/tasks.py

```diff
@@ -183,7 +184,10 @@ def get_grade_summary(user_id, course):
     """
     Return the grade for the given student in the addressed course.
     """
-    return CourseGradeFactory().create(User.objects.all().filter(id=user_id).first(), course).summary
+    try:
+        return CourseGradeFactory().create(User.objects.all().filter(id=user_id).first(), course).summary
+    except PermissionDenied:
```

Comment:
Why does this exception could appear?

Reply:
This code from the previous commit. It handle a situation when the user loses his access to the course block.